### PR TITLE
fix issues-557 issues-614

### LINKED ISSUE #557: conflict between @Service and @transactional 还需要其他设置？
hi:
      我在 dubbo the version 2.5.4-SNAPSHOT master 分支看到提交号为 3945b83956debcf2127606ef6d88a6d29a080094 已经解决了@Service and @transactional冲突问题. 但我在使用的时候，provider生成到SpringProxy这个类命名的地址下，我使用的是spring boot 1.5.2.RELEASE. 下面是我的代码，是否还需要设置其他配置:

```javascript
@Service(protocol = "dubbo", timeout = 20000, retries = -1)
@Transactional
public class GoodServiceImpl implements GoodService {
....
}
```

```javascript
<dependency>
     <groupId>com.alibaba</groupId>
     <artifactId>dubbo</artifactId>
     <version>2.5.4-SNAPSHOT</version>
	<exclusions>
            <exclusion>
                 <artifactId>spring</artifactId>
		 <groupId>org.springframework</groupId>
	    </exclusion>
	</exclusions>
</dependency>
```

- `[DUBBO] Register: dubbo://192.168.2.17:28801/org.springframework.aop.SpringProxy?.....
`

### PR DESCRIPTION
**Pull request by bohrqiu**
## What is the purpose of the change

fix #557 #614

### CODE REVIEW

File: dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/AnnotationBean.java [hunk 1]
```diff
@@ -136,19 +137,23 @@ public void destroy() {
     @Override
     public Object postProcessAfterInitialization(Object bean, String beanName)
         throws BeansException {
-        if (!isMatchPackage(bean)) {
+        Class targetClass=bean.getClass();
+        if(AopUtils.isAopProxy(targetClass)){
```
Comment (diecui1202) [inline dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/AnnotationBean.java]: Maybe you need to consider isCglibProxy or isJdkDynamicProxy ?

File: dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/AnnotationBean.java [hunk 2]
```diff
@@ -136,19 +137,23 @@ public void destroy() {
     @Override
     public Object postProcessAfterInitialization(Object bean, String beanName)
         throws BeansException {
-        if (!isMatchPackage(bean)) {
+        Class targetClass=bean.getClass();
+        if(AopUtils.isAopProxy(targetClass)){
+            targetClass=AopUtils.getTargetClass(bean.getClass());
```
Comment (diecui1202) [inline dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/AnnotationBean.java]: AopUtils.getTargetClass take a bean object as parameter ?

File: dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/AnnotationBean.java [hunk 3]
```diff
@@ -203,10 +208,14 @@ public Object postProcessAfterInitialization(Object bean, String beanName)
     @Override
     public Object postProcessBeforeInitialization(Object bean, String beanName)
         throws BeansException {
-        if (!isMatchPackage(bean)) {
+        Class targetClass=bean.getClass();
+        if(AopUtils.isAopProxy(targetClass)){
+            targetClass=AopUtils.getTargetClass(bean.getClass());
+        }
```
Comment (diecui1202) [inline dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/AnnotationBean.java]: Make sure the code format is right pls.

**Comment by diecui1202** [comment]
@bohrqiu Please check the review comments. It's recommend that add some unit test cases for this problem.

File: dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/AnnotationBean.java (hunk 1 above)
Comment (carryxyh) [reply dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/AnnotationBean.java]: That is a different code style.
Maybe format your code first.

**Pull request closed without merging**